packages/fx-core/src/plugins/resource/dotnet/index.ts

```diff
@@ -0,0 +1,232 @@
+      return err(returnSystemError(new Error("no capability"), this.name, "NoCapability"));
+    }
+
+    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
+    const currCap = solutionSettings.capabilities;
+    const pluginCtx = {
+      plugins: solutionSettings.activeResourcePlugins ?? [],
+      capabilities: [...currCap, ...newCap],
+    };
+
+    if (!currCap.includes(TabOptionItem.id) && !currCap.includes(BotOptionItem.id)) {
+      const webAppBicep = await this.generateWebAppBicep(pluginCtx);
+      result.push(webAppBicep);
+    }
+
+    if (newCap.includes(BotOptionItem.id) && currCap.includes(TabOptionItem.id)) {
+      const updateWebAppBicep = await this.updateWebAppBicep(pluginCtx);
+      result.push(updateWebAppBicep);
+    }
+
+    if (newCap.includes(BotOptionItem.id) && !currCap.includes(BotOptionItem.id)) {
+      const botBicep = await this.generateBotServiceBicep(pluginCtx);
+      result.push(botBicep);
+    }
+
+    ctx.logProvider.info(`[${this.name}] ${LogMessage.endGenerateResourceTemplate(newCap)}.`);
+    return ok(result);
+  }
+
+  @hooks([CommonErrorHandlerMW({ telemetry: { component: BuiltInFeaturePluginNames.dotnet } })])
+  async updateBicep(
+    ctx: v3.ContextWithManifestProvider,
+    inputs: v3.UpdateInputs
+  ): Promise<Result<v3.BicepTemplate[], FxError>> {
+    ctx.logProvider.info(`[${this.name}] ${LogMessage.startUpdateResourceTemplate}`);
+    const capabilities = this.getCapabilities(inputs);
+    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
+    const pluginCtx = {
+      plugins: solutionSettings.activeResourcePlugins ?? [],
+      capabilities: capabilities,
+    };
+    const result = await this.updateWebAppBicep(pluginCtx);
+
+    ctx.logProvider.info(`[${this.name}] ${LogMessage.endUpdateResourceTemplate}`);
+    return ok([result]);
+  }
+
+  private async generateWebAppBicep(pluginCtx: {
+    plugins: string[];
+    capabilities: string[];
+  }): Promise<v3.BicepTemplate> {
+    const webappTemplatePaths = [
+      PathInfo.webappProvisionOrchestrationPath,
+      PathInfo.webappProvisionModulePath,
+      PathInfo.webappConfigOrchestrationPath,
+      PathInfo.webappConfigModulePath,
+    ];
+    const bicepSnippets = await Promise.all(
+      webappTemplatePaths.map((path) => generateBicepFromFile(path, pluginCtx))
+    );
+
+    const result: v3.BicepTemplate = {
+      Provision: {
+        Orchestration: bicepSnippets[0],
+        Modules: { webapp: bicepSnippets[1] },
+      },
+      Configuration: {
+        Orchestration: bicepSnippets[2],
+        Modules: { webapp: bicepSnippets[3] },
+      },
+      Reference: WebappBicep.Reference,
+    };
+    return result;
+  }
+
+  private async updateWebAppBicep(pluginCtx: {
+    plugins: string[];
+    capabilities: string[];
+  }): Promise<v3.BicepTemplate> {
+    const configModule = await generateBicepFromFile(PathInfo.webappConfigModulePath, pluginCtx);
```

Comment:
Just a  question, why this function named updateXXX, but the logic is call generateXXX and return the result

Reply:
Good question.